### ISSUE TITLE
chopping sounds with AudioStreamTrack

### ISSUE THREAD

**Issue by sylyt62** [aiortc/aiortc#1072]
Wanna know if it is possible to play sounds in a different frame rate? 

I found that it is hard coded here, the audio packet time is set to be 0.02s (50fps): https://github.com/aiortc/aiortc/blob/e9c13eab915ddc27f365356ed9ded0585b9a1bb7/src/aiortc/mediastreams.py#L13

I try to change the way calculating timestamps, but the sound is skipping. My class looks like this: 
(the origin .wav file has 16k sample rate.)

```
class AudioTrack(AudioStreamTrack):
    kind = "audio"
    _start: float
    _timestamp: int

    def __init__(self, aud_path, sample_rate=16000, fps=25):
        super().__init__()  # don't forget this!
        self.wav = wave.open(aud_path, 'rb')
        self.sample_rate = sample_rate
        self.fps = fps
        self.chunk_size = round(sample_rate / fps)  # 640

    async def recv(self) -> Frame:
        if self.readyState != "live":
            raise MediaStreamError

        audio_data = self.wav.readframes(self.chunk_size)
        audio_data_np = np.frombuffer(audio_data, dtype=np.int16)
        if self.wav.getnchannels() == 2:
            audio_data_np = audio_data_np.reshape(-1, self.wav.getnchannels())
            audio_data_np = np.mean(audio_data_np, axis=1).astype(np.int16)
        audio_data_np = np.expand_dims(audio_data_np, 0)

        if hasattr(self, "_timestamp"):
            self._timestamp += self.chunk_size
            wait = self._start + (self._timestamp / self.sample_rate) - time.time()
            await asyncio.sleep(wait)
        else:
            self._start = time.time()
            self._timestamp = 0

        frame = AudioFrame.from_ndarray(array=audio_data_np, format="s16", layout="mono")
        frame.pts = self._timestamp
        frame.sample_rate = self.sample_rate
        frame.time_base = fractions.Fraction(1, self.sample_rate)

        return frame

```

**Comment by sylyt62** [comment]
I tried MediaPlayer() as well, it can play the .wav file properly. 
I looked into the code, and found out that there's a resampler inside.https://github.com/aiortc/aiortc/blob/e9c13eab915ddc27f365356ed9ded0585b9a1bb7/src/aiortc/contrib/media.py#L99

But in my use case, the audio data is sent in streaming (one chunk for one video frame), so I cannot use MediaPlayer.

**Comment by sylyt62** [comment]
Solved. 

@jlaine Seems like aiortc wouldn't create sdp offer according to specific tracks. I added an audio track with sample rate 16k and ptime 40ms to peerconnection, but the generated offer still contains the default atributes:

> m=audio 61921 UDP/TLS/RTP/SAVPF 96 0 8
c=IN IP4 2.0.0.1
a=sendrecv
a=extmap:1 urn:ietf:params:rtp-hdrext:sdes:mid
a=extmap:2 urn:ietf:params:rtp-hdrext:ssrc-audio-level
a=mid:1
a=msid:7efcc2eb-a421-41b5-8aed-dd5914c45052 b3c4a814-0993-42bb-a5d6-fb32d37c902d
a=rtcp:9 IN IP4 0.0.0.0
a=rtcp-mux
a=ssrc:2645354522 cname:eee68eb0-1a58-49c2-9fe7-a6ab3c3db647
a=rtpmap:96 opus/48000/2
a=rtpmap:0 PCMU/8000
a=rtpmap:8 PCMA/8000

I simply changed the line:

```
def change_sdp(sdp):
    target = 'a=rtpmap:96 opus/48000/2'
    update = 'a=rtpmap:96 opus/16000/1\r\na=fmtp:96 minptime=40;useinbandfec=1'
    return sdp.replace(target, update)
```
    
And the sound can be played properly.
